marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx

```diff
@@ -390,6 +398,79 @@ describe('RTL Source-to-entity map tests', () => {
          */
     })
 
+    test('CollapseAll/Expand All feature in JSON Source data table and Entity table', () => {
+
+        const { getByTestId, getByText, debug, queryByText } = render(<SourceToEntityMap {...data.mapProps}
+            mappingVisible={true}
+        />);
+
+        /* Validate collapse-expand in source table */
+        //Check if the expected source table elements are present in the DOM before hittting the Expan/Collapse button
+        expect(queryByText('suffix')).not.toBeInTheDocument();
+        expect(getByText('nutFreeName')).toBeInTheDocument();
+        expect(getByText('FirstNamePreferred')).toBeInTheDocument();
+        expect(getByText('LastName')).toBeInTheDocument();
+
+        let expandCollapseBtn = getByTestId('expandCollapseBtn-rowKey');
+
+        expect(expandCollapseBtn.textContent).toBe('Expand All'); // Validating the button label 
+
+        fireEvent.click(expandCollapseBtn); //Expanding all nested levels
+        expect(expandCollapseBtn.textContent).toBe('Collapse All'); // Validating the button label 
+        expect(getByText('suffix')).toBeInTheDocument();
```

Comment:
can we add an assertion for indentation check. Something similar to this used in this test file:
expect(getByText('suffix')).toHaveStyle("line-height: 2vh; text-indent: 20px;"); as applicable

Reply:
Done